HW2/P5/P5.txt

```diff
@@ -0,0 +1,29 @@
+###################### 1 Thread ########################
+Subproblem 1: replace range with prange:
+13.4102344219 average simulation frames per second with 1000 iterations
+
+Subproblem 2: spatial decomposition:
+526.000628065 average simulation frames per second with 1000 iterations
+
+Subproblem 3: spatial coherent sorting:
+572.582165363 average simulation frames per second with 1000 iterations
+
+Subproblem 4: Locking:
+461.008823972 average simulation frames per second with 1000 iterations
+
+
+
+###################### 4 Threads ########################
+Subproblem 1: replace range with prange:
+19.4908942205 average simulation frames per second with 1000 iterations
+
+Subproblem 2: spatial decomposition:
+882.601256482 average simulation frames per second with 1000 iterations
+
+Subproblem 3: spatial coherent sorting:
+959.56498753 average simulation frames per second with 1000 iterations
+
+Subproblem 4: Locking:
+744.779292039 average simulation frames per second with 1000 iterations
```

Comment:
I observed some more significant differences between 3 and 4 when using 4 threads.